src/components/UpgradeTableOfContents.tsx

```diff
@@ -70,54 +70,44 @@ interface IPropsItemsList {
   maxDepth: number
 }
 
-const ItemsList: React.FC<IPropsItemsList> = ({
-  items,
-  depth,
-  maxDepth,
-}: {
-  items: Array<Item>
-  depth: number
-  maxDepth: number
-}) => {
+const ItemsList: React.FC<IPropsItemsList> = ({ items, depth, maxDepth }) => {
   if (depth > maxDepth || !items) {
     return null
   }
-
   return (
     <>
       {items.map((item, index) => (
-        <ListItem margin={0} key={index}>
-          <TableOfContentsLink depth={depth} item={item} />
+        <ListItem m={0} key={index}>
+          <TableOfContentsLink item={item} />
         </ListItem>
       ))}
     </>
   )
 }
 
-export interface IProps {
-  items: Array<Item>
+function UpgradeTableOfContents(props: {
+  items: Array<{ id: string; title: string }>
+}): JSX.Element
+function UpgradeTableOfContents(props: {
+  maxDepth: number
+  items: Array<TableOfContentsItem>
+}): JSX.Element
+function UpgradeTableOfContents(props: {
```

Comment:
To type guard for both instances of incoming data objects, and ensure no other other object type can freely be passed in.